Fix remove unnecessary overview headings (628) (#632)
astronomer/docs: software/secrets-backend.md

## Secrets Backend Tool Integration Benefits

Integrating a secrets backend tool with Astronomer Software allows you to:

- Store Airflow variables and connections in a centralized location alongside secrets from other tools and systems used by your team, including Kubernetes secrets, SSL certificates, and more.
- Comply with internal security postures and policies that protect your organization.
- Recover in the case of an incident.
- Automatically pull Airflow variables and connections that are already stored in your secrets backend when you create a new Deployment instead of having to set them manually in the Airflow UI.

> Astronomer Software supports integration with a variety of secrets backend tools including:

Reviewer: ```suggestion
Astronomer Software integrates with the following secrets backend tools:
```

same comment as on Astro version of this doc

- Hashicorp Vault
- AWS Systems Manager Parameter Store
- Google Cloud Secret Manager
- Azure Key Vault

> Secrets backend integrations are applied individually to each Astronomer Software Deployment.

Reviewer: ```suggestion
Secrets backend integrations are configured individually with each Astronomer Software Deployment.
```
